lib/graphql/union_type.rb

```diff
@@ -57,14 +57,16 @@ def possible_types=(new_possible_types)
     end
 
     # @return [Array<GraphQL::ObjectType>] Types which may be found in this union
-    def possible_types
+    def possible_types(ctx=nil)
       @clean_possible_types ||= begin
         if @dirty_possible_types.respond_to?(:map)
           @dirty_possible_types.map { |type| GraphQL::BaseType.resolve_related_type(type) }
         else
           @dirty_possible_types
         end
       end
+
+      ctx ? filter_possible_types(@clean_possible_types, ctx) : @clean_possible_types
```

Comment:
`filter_possible_types` already has a similar conditional that defaults to returning `@clean_possible_types`. Not sure if it's preferred to keep this conditional here and make it more explicit what happens when there's no `ctx`, or just return the result of `filter_possible_types(@clean_possible_types, ctx)` here.

Reply:
Yah I removed the possiblity of a nil context in the update I am pushing. I am changing it to `NullContext` so it can still be operated on like a hash without erroring.